### PR TITLE
PP forward kwargs question

### PR DESCRIPTION
**Pull request by nlpfollower2**
Hey!

I've been exploring your fork and saw you're adding support for the llava model -- very cool. I was also adding some flavor to this awesome repo, though only adapting it for post-training losses like dpo, and adding stuff like attention mask, sample packing, eval.

What led me to your fork was this line:

```
pp_schedule.step(target=targets, losses=losses, **{"inputs_embeds":inputs_embeds})
```

I was trying to pass an attention mask to the forward step, and using kwargs seems like the right approach. However, I found that the torch stage implementation has this tendency to flatten kwargs and add them to the inputs for gradient computation. Which seems problematic.

Maybe, do you know what's the correct way to pass no-grad forward kwargs to the torch pipeline schedule step? Thanks a lot!

PS: I opened a PR from my "main" alt earlier but the commit log was crazy so reopening here.

### CODE REVIEW

**Comment by nlpfollower2** [comment]
Just figured it out. After some digging, monkey patching torch distributed ultimately got training to work with schedules like 1F1B and Interleaved1F1B; though not ZB-V for some reason. 

Anyway, sorry to bother. I'll be closing this. Best of luck in your project!
